unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/Tests/ChatChannelWindowMock.cs

```diff
@@ -0,0 +1,51 @@
+using DCL.Chat.HUD;
+using NSubstitute;
+using System;
+using UnityEngine;
+
+public class ChatChannelWindowMock : MonoBehaviour, IChatChannelWindowView
```

Comment:
Why do we need this? I dont see anything inside that cannot be mocked using NSubstitute.

Reply:
`TaskbarHUDShould` contains tests that should be completely redesigned. One of them is `ToggleWindowsProperly`. Its more like an integration test that brings many kind of issues when making changes in the HUD controllers. The test asserts the visibility/IsActive state of the view generated by its own controllers. So i do need a real behaviour in the background so the test makes sense. Anyway, will try to find a way to avoid the stub, replacing it by a Substitute instance, but im not sure about the results.

Reply:
```
var go = new GameObject("_channelHolder");
var channel = Substitute.For<IChatChannelWindowView>();
channel.When(x => x.Dispose()).Do(x => Destroy(go));
channel.When(x => x.Hide()).Do(x => go.SetActive(false));
channel.When(x => x.Show()).Do(x => go.SetActive(true));
```

Could this work?

Reply:
Yep, did the refactor to all the controller/views to use Substitute mocks instead of custom stubs

Reply:
Nice!